### ISSUE TITLE
A complaint for the children: what about learning disabilities?

### ISSUE THREAD

**Issue by PantsStatusZero** [a11yproject/a11yproject.com#763]
## Summary
There is a recurring issue in list and attempts to make software available to people with disabilities, which sadly a11y also falls victim to. Which is that learning disabilities are a class of disability and people with learning disabilities use computers and therefore the internet. 

### Major Follies
> The lack of acknowledgement that text-to-speech software is an accessibility tool or the belief that screen readers are a valid substitution or a replacement for text-to-speech software. 

Screen readers are designed with visually impaired people in mind. The default settings for most screen readers, at least in my experience, are overwhelming if you just have a learning disability and you just want to have a single paragraph read back to you rather then the whole screen or were ever you mouse cursor just happens to be. I personally need to hear myself talk when I am writing to get down what I am thinking on paper. That's really hard to do if your computer is randomly talking at you because you randomly bumped your desk. Noise Distraction is often a barrier for people with learning disabilities.

> Web site features that prevent the use of text-to-speech or otherwise interfere with the use of text-to-speech software 

- Disabling the ability to copy text and/or removing the copy option from menus. If you must disable copying of text an audio version for the content should be made available. Under no circumstance should copying and pasting be disabled for a text area that accepts user input. The inability to paste also interferes with speech-to-text software. 

- Manipulating the default copying behavior to limit the amount of characters that can be copied at once or in total or manipulating the default copying behavior to concatenate attribution\citation.

- Auto playing ads or video when the video is not the main content or is unrelated to the main content. This can sometime make a website unusable if you rely on text-to-speech.

- Layouts where a single paragraph is split into two with the use of an inline image with a caption. This often is mid-sentence or mid-word.

> Disabling spell check or not making spell check available. We should strive to make contextual spell check the norm. 

> The lack of ability to record or screen capture during interactive experiences. Multiple types of learning disabilities make it difficult to take notes. 

> The inability to note that you made need extra time to type a response when interacting with companies that use chat support. Chat support systems that kick you if you take to long to respond, even when you are still typing.

**Comment by ericwbailey** [comment]
Hi @PantsStatusZero, is this intended to be a post?

**Comment by PantsStatusZero** [comment]
Yes, I think it's a valid bug. You could consider it has a further question about the completeness of the web accessibility checklist.  

    On Friday, April 12, 2019, 8:05:22 PM CDT, Eric Bailey <notifications@github.com> wrote:  
 
 
Hi @PantsStatusZero, is this intended to be a post?

**Comment by ericwbailey** [comment]
Got it. What sources are you quoting for those blockquotes, and how would you intend these observations being worked into the checklist?

I agree that cognitive concerns are an important part, and one that we're trying to incorporate into the redesign.

**Comment by PantsStatusZero** [comment]
The quotes are mostly a first hand account, being a tech savvy person with a learning disability who has used text-to-speech and speech-to-text software on the internet. That said I could find supporting documentation on which disabilities, typical accommodations those relate to, and a couple real world examples of where those things have been an issue; if needed. 

On how to in corporate them into the current check list?

Most of these suggestions could live in the Forms, Media, or Testing sections or sub-sections of those. 

I'll also will try to find if there any good open source contextual spell check libraries out there in the wild for the Patterns page.

**Comment by ericwbailey** [comment]
That would be fantastic! I think firsthand experience is a great thing to have, as well as cited sources. 

Cognitive disability is something that I personally think is incredibly important to make people aware of, but I'm not too familiar with the space. So apologies if the tone seemed dismissive, I just wanted to make sure where you were coming from and that we'd be doing our due diligence by pointing to reputable sources.

I'd love to incorporate any suggestions you provide, or work with you if you don't want to submit a PR. In addition, is this something you'd be interested in writing about as a post on the site? I'd love an opportunity to publish more on the topic.

**Comment by sebacicltd** [comment]
Can I point you here...

Readability wiki

https://readabilityguidelines.myxwiki.org/xwiki/bin/view/Main/

**Comment by ericwbailey** [comment]
Closing for inactivity.
